assembl/static2/js/app/pages/threadAdmin.jsx

```diff
@@ -1,9 +1,97 @@
 import React from 'react';
+import { compose, graphql } from 'react-apollo';
+import { connect } from 'react-redux';
+import { get } from '../utils/routeMap';
+import ExportSection from '../components/administration/exportSection';
+import Navbar from '../components/administration/navbar';
+import DiscussionPreferenceLanguage from '../graphql/DiscussionPreferenceLanguage.graphql';
 
 class ThreadAdmin extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      exportLocale: null,
+      refetching: false,
+      translate: false
+    };
+  }
+
+  componentDidMount() {
+    this.props.router.setRouteLeaveHook(this.props.route, this.routerWillLeave);
+  }
+
+  componentWillUnmount() {
+    this.props.router.setRouteLeaveHook(this.props.route, null);
+  }
+
+  handleExportLocaleChange = (locale) => {
```

Comment:
Bind these methods as well

Reply:
Arrow functions and class properties like this are automatically binded to this @ayyazdaniaryan

Reply:
Since we're using Flow, we should use arrow function?

Reply:
yes